libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -775,6 +775,9 @@ function parseRequest(req: WebRequest): Promise<Activity> {
         function returnActivity(activity: Activity): void {
             if (typeof activity !== 'object') { throw new Error(`BotFrameworkAdapter.parseRequest(): invalid request body.`); }
             if (typeof activity.type !== 'string') { throw new Error(`BotFrameworkAdapter.parseRequest(): missing activity type.`); }
+            if (activity.timestamp && typeof activity.timestamp === 'string') { activity.timestamp = new Date(activity.timestamp); }
```

Comment:
You don't need the first test. You can just do "if (typeof activity.timestamp === 'string')"